stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -1069,6 +1069,28 @@ public class ChatClient internal constructor(
         return updateUsers(listOf(user)).map { it.first() }
     }
 
+    /**
+     * Updates specific user fields retaining the custom data fields which were set previously.
+     *
+     * @param id user ids
+     * @param set the key-value data which will be added to the existing user object
+     * @param unset the list of fields which will be removed from the existing user object
+     *
+     * @return executable async [Call]
+     */
+    @CheckResult
+    public fun partialUpdateUser(
+        id: String,
+        set: Map<String, Any> = emptyMap(),
+        unset: List<String> = emptyList(),
+    ): Call<User> {
+        return api.partialUpdateUser(
```

Comment:
What about add guard checks? We need to have user being set, we need to have set user with id that equals to argument. Can be platform side verification

Reply:
I'm not sure if we should do them client-side. The server error is meaningful and we'll have to do additional changes in our SDK if this requirement changes on the server side.

Reply:
I'll add a check